### ISSUE TITLE
JMS Connector IllegalArgumentException on message ack.

### ISSUE THREAD

**Issue by Traivor** [smallrye/smallrye-reactive-messaging#1622]
I'm getting the following stack trace when trying to use the JMS connector:

```
2022-02-17 04:25:01,077 ERROR [io.smallrye.reactive.messaging.provider] (pool-15-thread-1) SRMSG00201: Error caught while processing a message: java.lang.IllegalArgumentException
        at deployment.collectbot.war//io.smallrye.reactive.messaging.jms.IncomingJmsMessage.lambda$ack$0(IncomingJmsMessage.java:116)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

This corresponds to:
```
    @Override
    public CompletionStage<Void> ack() {
        return CompletableFuture.runAsync(() -> {
            try {
                delegate.acknowledge();
            } catch (JMSException e) {
                throw new IllegalArgumentException();
            }
        }, executor);
    }
```

Since the IllegalArgumentException doesn't include the original exception, I have nothing to go on for what's happening.

If I switch to using JMS directly, including doing the message.acknowledge(), things work fine. No change to the underlying JMS configuration in the application server.

I'm on Wildfly 26, default InVm connection factory, Java 11. Messaging 3.6.0 or 3.14.0.

Any thoughts?

**Comment by Traivor** [comment]
I'm afraid not. I'm so slammed I don't know when I'd have time to get a debugger going. If there's a release that doesn't swallow the exception I can certainly give it another look.

For background, my goal is to have channel transport be user configurable, with the default for a single app instance deployment being something in-vm. For a multi instance, ultimately multi-regional, the user would configure something else. I had been using the vertx-eventbus connector for my default with success, but it seems that has been removed from the 3.x line.

**Comment by cescoffier** [comment]
Fixed by  #1630 - at least we will get the reason.

**Comment by Traivor** [comment]
Cool. Thanks!
